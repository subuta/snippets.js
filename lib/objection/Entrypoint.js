import { build, format, snippets as s } from 'bld.js'
import pluralize from 'pluralize'
import _ from 'lodash'

export default (Models = {}) => {
  const keys = _.map(_.keys(Models), (r) => s.modelName(r))
  const modelImports = _.zip(_.map(keys, (k) => `./${k}`), keys)
  const models = _.zipObject(keys, keys)

  const imports = s.import([
    ['lodash', '_'],
    ['objection', null, [
      'Model'
    ]],
    ['src/api/utils/knex', 'knex'],
    ...modelImports
  ])

  return build`
    ${imports}
    
    // assign connection to knex.
    Model.knex(knex)
    
    // then require all without itself.
    const modules = ${s.raw(models)}
    
    // pick class definition from modules.
    const models = _.transform(modules, (result, module, key) => result[key] = module.default || module, {})
    
    // call register of each model.
    _.each(modules, (fn) => fn.register(models))
    
    ${s.export(build`
      models
    `)}
  `
}
