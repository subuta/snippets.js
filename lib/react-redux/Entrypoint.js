import { build, format, snippets as s } from 'bld.js'
import _ from 'lodash'
import pluralize from 'pluralize'

export default () => {
  const imports = s.import([
    ['react', 'React'],
    ['react-dom', 'ReactDOM'],
    ['react-redux', null, [
      ['Provider']
    ]],
    ['./store', 'store'],
    ['./components/Routes', 'Routes'],
    ['./style', 'withStyles'],
    ['recompose', null, [
      ['compose']
    ]]
  ])

  return build`
    ${imports}
    
    const enhance = compose(
      withStyles
    )
    
    let App = enhance(() => {
      return (
        <div>
          <Provider store={store}>{Routes}</Provider>
        </div>
      )
    })
    
    let render = () => {
      const appNode = document.getElementById('app')
      ReactDOM.render(<App />, appNode)
    }
    
    // Native
    // Check if the DOMContentLoaded has already been completed
    if (document.readyState === 'complete' || document.readyState !== 'loading') {
      render()
    } else {
      document.addEventListener('DOMContentLoaded', render)
    }
    
    // make browserify-hmr work.
    if (module.hot) {
      module.hot.accept()
    }

  `
}