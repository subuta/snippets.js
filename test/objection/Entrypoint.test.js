/* global expect, describe, it */

import { build, format, snippets as s } from 'bld.js'

import Entrypoint from 'lib/objection/Entrypoint'

describe('objection/Entrypoint', () => {
  it('should create objection Entrypoint', () => {
    expect(format(Entrypoint({modelDir: 'src/model'}))).toMatchSnapshot()
  })
})
