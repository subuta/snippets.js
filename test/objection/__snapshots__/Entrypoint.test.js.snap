// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`objection/Entrypoint should create objection Entrypoint 1`] = `
"require('babel-register')

import _ from 'lodash'
import requireGlob from 'require-glob'
import {Model} from 'objection'
import {absolutePath} from '../../config'
import knex from 'src/utils/knex'

// assign connection to knex.
Model.knex(knex)

// then require all without itself.
const modules = requireGlob.sync([
  absolutePath('src/model/*.js'),
  \`!\${absolutePath('src/model/index.js')}\`,
  \`!\${absolutePath('src/model/Model.js')}\`
])

// pick class definition from modules.
const models = _.transform(
  modules,
  (result, module, key) => (result[key] = module.default || module),
  {}
)

// call register of each model.
_.each(modules, (fn) => fn.register(models))

export default models
"
`;
